go/vt/mysqlctl/backup.go

```diff
@@ -134,9 +149,13 @@ func Backup(ctx context.Context, params BackupParams) error {
 	if err != nil {
 		return vterrors.Wrap(err, "failed to find backup engine")
 	}
-
 	// Take the backup, and either AbortBackup or EndBackup.
-	usable, err := be.ExecuteBackup(ctx, params, bh)
+	beParams := params.Copy()
```

Comment:
This is a bit awkward, and copying structs makes me uncomfortable. Is there a way to lint for exhaustive field copying?

Reply:
just thinking out loud here.. can we have tow stats , enginestats and storagestats in backupparam.. so you won't end up copy

Reply:
I think if we did that approach we might end up needing three stats, one for the engine, one for the storage, and one for the controlling function (`mysqlctl.Backup`).

Overall I think I am OK with the copy, and one thing I didn't know about Golang when I wrote the comment above is that this kind of struct creation is exhaustive:

	a1 := A{
		"hello",
		true,
		3,
	}

It won't let you omit any struct fields. So I at least feel good about that. The only risk now is if someone swaps the order of two struct fields that have the same type 😬